src/app/core/allocation/allocation-preview.service.ts

```diff
@@ -11,23 +11,39 @@ export class AllocationPreviewService {
 
   constructor(private augury: AuguryService) {}
 
-  getAllocationPreview({ set_inventory_uri, name, startAt, endAt, totalGoal, dailyMinimum, zones }): Observable<AllocationPreview> {
-    return this.augury
-      .follow('prx:flight-allocation-preview', {
+  getAllocationPreview({ id, set_inventory_uri, name, startAt, endAt, totalGoal, dailyMinimum, zones }): Observable<AllocationPreview> {
+    let preview: HalObservable<HalDoc>;
+    if (id) {
```

Comment:
if there is a flight id, follow the preview on the flight, otherwise follow the "unsaved" preview on the API root